ui/src/views/Settings.vue

```diff
@@ -106,10 +326,54 @@ export default {
     clearInterval(this.urlCheckInterval);
     next();
   },
-  created() {
-    this.getConfiguration();
-  },
   methods: {
+    async listWidgetOptions() {
+      this.loading.getConfiguration = true;
+      const taskAction = "list-widget-options-for-UI";
+      const eventId = this.getUuid();
+
+      // register to task error
+      this.core.$root.$once(
+        `${taskAction}-aborted-${eventId}`,
+        this.listWidgetOptionsAborted
+      );
+
+      // register to task completion
+      this.core.$root.$once(
+        `${taskAction}-completed-${eventId}`,
+        this.listWidgetOptionsCompleted
+      );
+
+      const res = await to(
+        this.createModuleTaskForApp(this.instanceName, {
+          action: taskAction,
+          extra: {
+            title: this.$t("action." + taskAction),
+            isNotificationHidden: true,
+            eventId,
+          },
+        })
+      );
+      const err = res[0];
+
+      if (err) {
+        console.error(`error creating task ${taskAction}`, err);
+        this.error.listWidgetOptions = this.getErrorMessage(err);
+        this.loading.getConfiguration = false;
+        return;
+      }
+    },
+    listWidgetOptionsAborted(taskResult, taskContext) {
+      console.error(`${taskContext.action} aborted`, taskResult);
+      this.error.listWidgetOptions = this.$t("error.generic_error");
```

Comment:
Let's show a specific error message, such as "Cannot retrieve UI options". (I'd prefer to have two distinct actions for mail server list and timezones list, but that's ok)